File: aaaa/volume/DISK01/TESTLIB/TestF9Abend.java
import java.nio.file.*;
import java.nio.file.StandardOpenOption;

/**
 * Test F9 ABEND Scenario
 * This program simulates F9 key press and triggers CEE3204S ABEND
 */
public class TestF9Abend {
    
    public static void main(String[] args) {
        System.out.println("*** F9 ABEND TEST SCENARIO ***");
        System.out.println("Simulating F9 key press that causes ABEND");
        
        try {
            // Simulate F9 key press detection
            String keyPressed = "F9";
            System.out.println("[TEST] F9 key detected: " + keyPressed);
            
            // Trigger ABEND scenario
            triggerF9Abend();
            
        } catch (Exception e) {
            System.err.println("[ERROR] Test failed: " + e.getMessage());
            e.printStackTrace();
            System.exit(999);
        }
    }
    
    private static void triggerF9Abend() {
        System.err.println("[ABEND] *** CRITICAL ERROR - CEE3204S ABEND TRIGGERED ***");
        System.err.println("[ABEND] ABEND CODE: CEE3204S");
        System.err.println("[ABEND] DESCRIPTION: The system detected a protection exception (System Completion Code=0C4)");
        System.err.println("[ABEND] CAUSE: F9 key pressed - unsupported function key access");
        System.err.println("[ABEND] LOCATION: TestF9Abend.triggerF9Abend()");
        System.err.println("[ABEND] TIMESTAMP: " + new java.util.Date());
        System.err.println("[ABEND] SCENARIO: DevOps Auto-Fix Testing - F9 Key ABEND");
        
        // Log ABEND to file for Zabbix monitoring
        logAbendToFile();
        
        // Create Zabbix trigger file
        createZabbixTriggerFile();
        
        // Display detailed error information
        System.err.println("[ABEND] *** SYSTEM DETAILS ***");
        System.err.println("[ABEND] SEVERITY: CRITICAL");
        System.err.println("[ABEND] IMPACT: System halt - User session terminated");
        System.err.println("[ABEND] ACTION_REQUIRED: Automatic remediation via CI/CD pipeline");
        System.err.println("[ABEND] EXPECTED_FIX: Deploy fixed MAIN001.class with proper F9 handling");
        System.err.println("[ABEND] MONITORING_TRIGGER: Zabbix should detect this ABEND and trigger auto-fix");
        
        System.err.println("[ABEND] *** SYSTEM DUMP INITIATED ***");
        System.err.println("[ABEND] *** PROGRAM TERMINATED ABNORMALLY ***");
        System.err.println("[ABEND] *** ZABBIX MONITORING SHOULD ACTIVATE AUTO-FIX PIPELINE ***");
        
        // Force abnormal termination with CEE3204S exit code
        System.exit(204); // Exit with CEE3204S code (204 = protection exception)
    }
    
    private static void logAbendToFile() {
        try {
            String logFile = "/home/aspuser/app/logs/abend.log";
            String timestamp = new java.text.SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(new java.util.Date());
            
            String logEntry = String.format(
                "[%s] ABEND CEE3204S in TestF9Abend.triggerF9Abend(): F9 key pressed - Protection exception in function key handler (Terminal: system, Session: test_session)%n",
                timestamp
            );
            
            // Ensure log directory exists
            java.io.File logDir = new java.io.File("/home/aspuser/app/logs");
            if (!logDir.exists()) {
                logDir.mkdirs();
            }
            
            // Append to log file
            Files.write(
                Paths.get(logFile), 
                logEntry.getBytes(), 
                StandardOpenOption.CREATE, 
                StandardOpenOption.APPEND
            );
            
            System.err.println("[ABEND] ABEND logged to: " + logFile);
            
        } catch (Exception e) {
            System.err.println("[ABEND] Failed to log ABEND: " + e.getMessage());
        }
    }
    
    private static void createZabbixTriggerFile() {
        try {
            String triggerFile = "/home/aspuser/app/logs/zabbix_trigger.log";
            String timestamp = new java.text.SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(new java.util.Date());
            
            String triggerEntry = String.format(
                "[%s] ZABBIX_TRIGGER ABEND=CEE3204S SCENARIO=F9_KEY_ABEND PROGRAM=TestF9Abend TERMINAL=system SESSION=test_session%n",
                timestamp
            );
            
            // Ensure log directory exists
            java.io.File logDir = new java.io.File("/home/aspuser/app/logs");
            if (!logDir.exists()) {
                logDir.mkdirs();
            }
            
            // Write trigger file
            Files.write(
                Paths.get(triggerFile), 
                triggerEntry.getBytes(), 
                StandardOpenOption.CREATE, 
                StandardOpenOption.APPEND
            );
            
            System.err.println("[ABEND] Zabbix trigger file created: " + triggerFile);
            System.err.println("[ABEND] Trigger content: " + triggerEntry.trim());
            
        } catch (Exception e) {
            System.err.println("[ABEND] Failed to create Zabbix trigger file: " + e.getMessage());
        }
    }
}

File: aaaa/volume/DISK01/TESTLIB/test_f9_abend.py
#\!/usr/bin/env python3
"""
Test F9 ABEND scenario with the restored ABEND version
"""

import subprocess
import os
import time
import threading

def create_f9_input():
    """Create F9 input file after 3 seconds"""
    time.sleep(3)
    with open("/tmp/asp_input_webui.txt", "w") as f:
        f.write("F9\n")
    print("[TEST] F9 input file created")

def test_f9_abend():
    """Test F9 ABEND scenario"""
    print("=== Testing F9 ABEND Scenario (CEE3204S) ===")
    print("")
    
    # Clear logs first
    for log_file in ["/home/aspuser/app/logs/abend.log", "/tmp/asp_input_webui.txt"]:
        if os.path.exists(log_file):
            if "abend.log" in log_file:
                with open(log_file, "w") as f:
                    f.write("")  # Clear ABEND log
            else:
                os.remove(log_file)
    
    # Start input thread
    input_thread = threading.Thread(target=create_f9_input)
    input_thread.start()
    
    print("Starting MAIN001 with ABEND version...")
    print("Expecting CEE3204S ABEND when F9 is pressed...")
    print("")
    
    try:
        process = subprocess.Popen(
            ["java", "-cp", ".:../../../server/java_classes", "MAIN001"],
            stdout=subprocess.PIPE,
            stderr=subprocess.PIPE,
            text=True
        )
        
        # Wait for process (should exit with code 204)
        stdout, stderr = process.communicate(timeout=15)
        exit_code = process.returncode
        
        print(f"=== Process Exit Code: {exit_code} ===")
        if exit_code == 204:
            print("SUCCESS: Process exited with CEE3204S code (204)")
        else:
            print(f"UNEXPECTED: Process exited with code {exit_code}")
        
        print("\n=== MAIN001 Output ===")
        print(stdout[-1500:])  # Last 1500 characters
        
        if stderr:
            print("\n=== MAIN001 ABEND Messages ===")
            print(stderr[-1000:])  # Last 1000 characters
            
    except subprocess.TimeoutExpired:
        print("Process timeout - killing...")
        process.kill()
        stdout, stderr = process.communicate()
        print("=== Timeout Output ===")
        if stdout:
            print(stdout[-1000:])
    
    except Exception as e:
        print(f"Error: {e}")
    
    # Wait for thread
    input_thread.join()
    
    print("\n" + "="*60)
    print("CHECKING ABEND LOGS")
    print("="*60)
    
    # Check ABEND log
    abend_log = "/home/aspuser/app/logs/abend.log"
    if os.path.exists(abend_log):
        with open(abend_log, "r") as f:
            content = f.read().strip()
            if content:
                print("ABEND LOG FOUND:")
                print(content)
                if "CEE3204S" in content and "F9" in content:
                    print("\nSUCCESS: F9 ABEND correctly logged for Zabbix monitoring\!")
                else:
                    print("\nWARNING: ABEND log exists but may not be F9-related")
            else:
                print("ABEND log exists but is empty")
    else:
        print("No ABEND log found")
    
    # Check Zabbix trigger log
    zabbix_log = "/home/aspuser/app/logs/zabbix_trigger.log"
    if os.path.exists(zabbix_log):
        print(f"\nZABBIX TRIGGER LOG:")
        with open(zabbix_log, "r") as f:
            lines = f.readlines()
            if lines:
                for line in lines[-3:]:
                    print(f"  {line.strip()}")
            else:
                print("  (empty)")

if __name__ == "__main__":
    test_f9_abend()
